### PR TITLE
[Merged by Bors] - chore: prepare for multi-goal linter

### PR DESCRIPTION
**Pull request by adomani**
Adaptations for #12339.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — COMMENTED** [review]
Thanks!
maintainer merge

File: Mathlib/Computability/TMToPartrec.lean [hunk 1]
```diff
@@ -1604,7 +1604,7 @@ def trStmts₁ : Λ' → Finset Λ'
 theorem trStmts₁_trans {q q'} : q' ∈ trStmts₁ q → trStmts₁ q' ⊆ trStmts₁ q := by
   induction q with
   | move _ _ _ q q_ih => _ | clear _ _ q q_ih => _ | copy q q_ih => _ | push _ _ q q_ih => _
-  | read q q_ih => _ | succ q q_ih => _ | pred q₁ q₂ q₁_ih q₂_ih => _ | ret => _
+  | read q q_ih => _ | succ q q_ih => _ | pred q₁ q₂ q₁_ih q₂_ih => _ | ret => _ <;>
   all_goals
```
Comment (kim-em) [inline Mathlib/Computability/TMToPartrec.lean:1608]: I don't understand what's going on here. Don't you want to remove the `all_goals`, if you're adding `<;>`?

File: Mathlib/Data/Seq/Computation.lean [hunk 2]
```diff
@@ -503,8 +503,8 @@ theorem length_thinkN (s : Computation α) [_h : Terminates s] (n) :
 
 theorem eq_thinkN {s : Computation α} {a n} (h : Results s a n) : s = thinkN (pure a) n := by
   revert s
-  induction n with | zero => _ | succ n IH => _
-  all_goals intro s; apply recOn s (fun a' => _) fun s => _ <;> intro a h
+  induction n with | zero => _ | succ n IH => _ <;>
+  intro s <;> apply recOn s (fun a' => _) fun s => _ <;> intro a h
```
Comment (kim-em) [inline Mathlib/Data/Seq/Computation.lean:507]: Could you use `x <;> (y; w; z)` here instead, to make the points at which goals bifurcate clearer? As is, you can't tell from the syntax that there is no further splitting.

**Comment by kim-em** [comment]
bors d+

File: Mathlib/Data/Seq/Computation.lean (hunk 2 above)
Comment (adomani) [reply Mathlib/Data/Seq/Computation.lean]: I modified this, I was not able to use `<;> [...]`, but I hope I made it clearer.

What is going on is that the first `intro <;> apply` is really an `intro; apply` for the initial 2 goals.  After the `apply` there are 4 goals left and the final `intro` applies to all.

File: Mathlib/Computability/TMToPartrec.lean (hunk 1 above)
Comment (adomani) [reply Mathlib/Computability/TMToPartrec.lean:1608]: Good catch!  I was convinced that I had tried and it failed, but it seems to work locally now.

**Review by adomani — COMMENTED** [review]
Kim, thank you for the quick review!  I clarified the syntax and will merge once CI is happy!

**Comment by adomani** [comment]
bors merge

**Pull request closed without merging**